### PR TITLE
Bookstore: Redux reducers and actions

### PR DESCRIPTION
**Pull request by romans-adi**
For this pull request:

## Reducers, actions, provider
- Added Redux Toolkit dependency for the project.
- Created a `store.js` file to configure the reducers in the store.
- Created two slices: `booksSlice.js` and c`ategoriesSlice.js`.
- Added three actions to the `booksSlice.js` to handle adding books, removing books, and selecting a category.
- In the `categoriesSlice.js`, set the initial state as 'Under construction' and the `checkStatus` reducer is empty since the value is initialized as a property in the `initial state`.
- Added the `Provider` component to the application in `App.js`.

## Category selection (additional requirement)

- Implemented the `selectBooksByCategory` function to filter the existing books based on the selected category.
- Updated the `AddBook.jsx` component to dynamically filter the list of selected categories (hardcoded). This change was necessary to allow the filtering action.

### CODE REVIEW

**Review by Ibizugbe — CHANGES_REQUESTED** [review]
Hi @romans-adi 👋👋, 

tryagain


Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights✅✅

- [x] Linters are correctly setup and they are passing👍.
- [x] Gitflow have been used👍.
- [x] PR is descriptive👍. 


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/categories/categoriesSlice.js [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+const categoriesSlice = createSlice({
+  name: 'categories',
+  initialState,
+  reducers: {
+    checkStatus: () => {
+      // initial state is set as an object property on line 5
+    },
+  },
+});
```
[the remark this answers is not in the document]
Comment (romans-adi) [reply src/redux/categories/categoriesSlice.js:16]: Thank you for the suggestion! I misunderstood the requirement, but now I understand the idea behind it. Have a nice day!

**Review by nedjwab — APPROVED** [review]
# 🟢 Project Approved! 🏆

## Hi  @romans-adi   👋

Your project is complete! ✔️✔ , I loved your work, it looks great 🔥🔥🔥🔥 🏁

Your code was very professional 💯

There is nothing else to say other than... _it's time to merge it_ :shipit:
Congratulations! 🎉

 
## Highlights 🌟

- ✔️ Descriptive commmit messsages
- ✔️ Descriptive PR description
- ✔️ No linter Errors

- The code is top-notch! 💯 and have a huge potential.

Cheers and Happy coding!👏👏👏

## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.


_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by romans-adi** [comment]
I greatly appreciate your review, @nedjwab!
Wishing you a pleasant evening! :fireworks:

**Pull request closed without merging**